node/src/proxy_server/mod.rs

```diff
@@ -1095,24 +1094,18 @@ impl IBCDHelper for IBCDHelperReal {
                 .dns_failure_retries
                 .insert(stream_key, dns_failure_retry);
         }
-
-        let tth_args = TryTransmitToHopperArgs {
-            main_cryptde: proxy.main_cryptde,
+        let retire_stream_key_sub_opt = if retire_stream_key {
+            Some(proxy.out_subs("ProxyServer").stream_shutdown_sub.clone())
```

Comment:
Not your fault, but `proxy` would be much better as `proxy_server`. Even `server` would be (a little) better, because "server" is more descriptive of its role than "proxy" is.